ports/stm32: implement TX/RX inversion parameter for UART on STM32H7

ennyKey opened the pull request:
On my board I am using a STM32H753 controller and needed the inversion feature for the UART which is not yet implemented in the STM32 port.
So I have implemented the missing keyword-only parameter `invert` for the `UART.init` function in accordance to the [docs](https://docs.micropython.org/en/latest/library/machine.UART.html#machine.UART).

- added constants `UART.INV_TX` and `UART.INV_RX` 
- added the new `invert` keyword parameter to the `uart_init` function.
- adapted the `uart_init` call in the main.c for the STM32H7 to the new function definition
- added `invert` setting to `uart_print` output

All changes apply for STM32H7 only. Due to define statements they have no effect for all other STM32 controllers.

I have tested the new Implementation on my STM32H753 succesfully.

dpgeorge commented on ports/stm32/machine_uart.c:
```diff
@@ -37,10 +37,21 @@
     { MP_ROM_QSTR(MP_QSTR_CTS), MP_ROM_INT(UART_HWCONTROL_CTS) }, \
     { MP_ROM_QSTR(MP_QSTR_IRQ_RXIDLE), MP_ROM_INT(UART_FLAG_IDLE) }, \
+    { MP_ROM_QSTR(MP_QSTR_INV_TX), MP_ROM_INT(UART_ADVFEATURE_TXINVERT_INIT) }, \
+    { MP_ROM_QSTR(MP_QSTR_INV_RX), MP_ROM_INT(UART_ADVFEATURE_RXINVERT_INIT) }, \
```
To reduce code duplication, please do this:
```c
#if defined(STM32H7)
#define MICROPY_PY_MACHINE_UART_INV_ENTRY \
    { MP_ROM_QSTR(MP_QSTR_INV_TX), MP_ROM_INT(UART_ADVFEATURE_TXINVERT_INIT) }, \
    { MP_ROM_QSTR(MP_QSTR_INV_RX), MP_ROM_INT(UART_ADVFEATURE_RXINVERT_INIT) },
#else
#define MICROPY_PY_MACHINE_UART_INV_ENTRY
#endif

#define MICROPY_PY_MACHINE_UART_CLASS_CONSTANTS \
    { MP_ROM_QSTR(MP_QSTR_RTS), MP_ROM_INT(UART_HWCONTROL_RTS) }, \
    { MP_ROM_QSTR(MP_QSTR_CTS), MP_ROM_INT(UART_HWCONTROL_CTS) }, \
    { MP_ROM_QSTR(MP_QSTR_IRQ_RXIDLE), MP_ROM_INT(UART_FLAG_IDLE) }, \
    MICROPY_PY_MACHINE_UART_INV_ENTRY \
```

dpgeorge commented on ports/stm32/machine_uart.c:
```diff
@@ -124,11 +135,26 @@ static void mp_machine_uart_print(const mp_print_t *print, mp_obj_t self_in, mp_
+        else if (cr2 & USART_CR2_RXINV) {
+            mp_print_str(print, "INV_RX");
+        }
+        #endif
```
Please use the same code pattern for printing RTS/CTS above.

dpgeorge commented on ports/stm32/machine_uart.c:
```diff
@@ -151,11 +178,18 @@ static void mp_machine_uart_init_helper(machine_uart_obj_t *self, size_t n_args,
     struct {
+        #if defined(STM32H7) 
+        mp_arg_val_t baudrate, bits, parity, stop, flow, timeout, timeout_char, rxbuf, read_buf_len, invert;
+        #else
```
You can just do:
```c
mp_arg_val_t baudrate, bits, parity, stop, flow, timeout, timeout_char, rxbuf, read_buf_len;
#if defined(STM32H7)
mp_arg_val_t invert;
#endif
```

dpgeorge commented on ports/stm32/uart.h:
```diff
@@ -82,8 +82,13 @@ extern const mp_irq_methods_t uart_irq_methods;
+    uint32_t baudrate, uint32_t bits, uint32_t parity, uint32_t stop, uint32_t flow, uint32_t invert);
+#else
 bool uart_init(machine_uart_obj_t *uart_obj,
     uint32_t baudrate, uint32_t bits, uint32_t parity, uint32_t stop, uint32_t flow);
```
I suggest to always accept the `invert` parameter to this C function, to simplify the callers of this function.  But only on H7 will it actually do anything.

dpgeorge reviewed (changes requested):
Thanks for the contribution, this is a nice addition that conforms to the existing docs.

(Note that inversion is also available on H5 MCUs, so this can eventually be extended to more MCUs in the future.)

Just a few minor comments about coding convention.  Also please check the CI and make sure everything passes.

Commit:
stm32/machine_uart: Implement TX/RX inversion parameter for UART on H7.

Implemented the keyword-only parameter `invert` for the `UART.init` method
in accordance with the docs:
- added constants `UART.INV_TX` and `UART.INV_RX`
- added the new `invert` keyword parameter to the `uart_init` function
- adapted the `uart_init` call
- added invert setting to `uart_print` output

The feature applies only to STM32H7.

Signed-off-by: ennyKey <ennyKey@fn.de>

dpgeorge reviewed (approved):
This looks good now, thanks for updating (and sorry it took ages to get back to this).
